include/picongpu/fields/incidentField/profiles/GaussianPulse.hpp

```diff
@@ -34,328 +34,345 @@
+                 *  @param x coordinate at which the polynomial is evaluated
                  */
-                template<typename T_Params>
-                struct GetFunctorIncidentB<profiles::GaussianPulse<T_Params>>
+                HDINLINE float_X simpleLaguerre(uint32_t const n, float_X const x) const
                 {
-                    using type = detail::ApproximateIncidentB<
-                        typename GetFunctorIncidentE<profiles::GaussianPulse<T_Params>>::type>;
-                };
-            } // namespace detail
-        } // namespace incidentField
-    } // namespace fields
-} // namespace picongpu
+                    // Result for special case n == 0
+                    if(n == 0)
+                        return 1.0_X;
+                    uint32_t currentN = 1;
+                    float_X laguerreNMinus1 = 1.0_X;
+                    float_X laguerreN = 1.0_X - x;
+                    float_X laguerreNPlus1(0.0_X);
+                    while(currentN < n)
+                    {
+                        // Core statement of the algorithm
+                        laguerreNPlus1 = ((2.0_X * float_X(currentN) + 1.0_X - x) * laguerreN
+                                          - float_X(currentN) * laguerreNMinus1)
+                            / float_X(currentN + 1u);
+                        // Advance by one order
+                        laguerreNMinus1 = laguerreN;
+                        laguerreN = laguerreNPlus1;
+                        currentN++;
+                    }
+                    return laguerreN;
+                }
+            };
+        } // namespace detail
+
+
+        template<typename T_Param>
+        struct GaussianPulseEnvelope : public detail::BaseParamUnitless<T_Param>
+        {
+            using Base = typename detail::BaseParamUnitless<T_Param>;
+            using Unitless = detail::BaseParamUnitless<T_Param>;
+
+            static constexpr float_X TIME_SHIFT = -Base::PULSE_INIT * Base::PULSE_DURATION;
+
+            HDINLINE static float_X getEnvelope(float_X const time)
+            {
+                auto const exponent = time / (2.0_X * Unitless::PULSE_DURATION);
+
+                return math::exp(-exponent * exponent);
+            }
+
+            HINLINE static std::string getName()
+            {
+                return "GaussianPulse";
+            }
+        };
+
+        template<typename T_Params, typename T_LongitudinalEnvelope>
+        struct GaussianPulse
+        {
+            using LongitudinalEnvelope = T_LongitudinalEnvelope;
+            //! Get text name of the incident field profile
+            HINLINE static std::string getName()
+            {
+                // This template is used for both Gaussian and PulseFrontTilt, distinguish based on tilt value
+                using TiltParam = detail::TiltParam<T_Params>;
+                bool isTilted = (std::abs(TiltParam::TILT_AXIS_1) + std::abs(TiltParam::TILT_AXIS_2) > 0);

```

Comment:
you could use your smarter method from above 
```suggestion
                bool isTilted = TiltParam::TILT_AXIS_1 || TiltParam::TILT_AXIS_2;

```

Reply:
Code that existed before